culprit_finder/src/culprit_finder/cli.py

```diff
@@ -0,0 +1,90 @@
+"""
+Command-line interface for the Culprit Finder tool.
+
+This module acts as the entry point for the application, handling argument parsing, input validation,
+and authentication checks. It initializes the `CulpritFinder` with user-provided parameters
+(repository, commit range, workflow) and reports the identified culprit commit or the lack thereof.
+"""
+
+import argparse
+import logging
+import os
+import sys
+
+from culprit_finder import culprit_finder
+from culprit_finder import github
+
+logging.basicConfig(
+  level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+
+
+def _validate_repo(repo: str) -> str:
+  parts = repo.split("/")
+  if len(parts) != 2 or not all(parts):
+    raise argparse.ArgumentTypeError(f"Invalid repo format: {repo}")
+
+  return repo
+
+
+def main() -> None:
+  """
+  Entry point for the culprit finder CLI.
+
+  Parses command-line arguments then initiates the bisection process using CulpritFinder.
+  """
+  parser = argparse.ArgumentParser(description="Culprit finder for GitHub Actions.")
+  parser.add_argument(
+    "--repo",
+    required=True,
+    help="Target GitHub repository (e.g., owner/repo)",
+    type=_validate_repo,
+  )
+  parser.add_argument("--start", required=True, help="Last known good commit SHA")
+  parser.add_argument("--end", required=True, help="First known bad commit SHA")
+  parser.add_argument(
+    "--workflow",
+    required=True,
+    help="Workflow filename (e.g., build_and_test.yml)",
+  )
+
```

Comment:
Can you add short flag alternatives for these parameters? E.g: `-r` for repo, `-w` for workflow, etc.